test/integration/integration_test.go

```diff
@@ -725,6 +730,22 @@ func succeedsOrFails(execute commandRunner, command string, expectedResult strin
 	return nil
 }
 
+func verifyHTTPResponseWithRetry(partOfResponse string, url string, urlSuffix string, assertion string, expected string) error {
+	var err error
+	delay := [3]int{100, 400, 2500}
+
+	for i := 0; i < len(delay); i++ {
+		err = verifyHTTPResponse(partOfResponse, url, urlSuffix, assertion, expected)
+		if err == nil {
+			break
+		}
+		fmt.Printf("HTTP check (%v) has failed, trying again in %v ms. Error: %v\n", i+1, delay[i], err)
+		time.Sleep(time.Millisecond * time.Duration(delay[i]))
+	}
+
+	return err
+}
```

Comment:
@agajdosi This should get exposed to the cucumber test steps i.e. when we read through tests it should be clear that we have some waiting time with n times of retry. Then it would be transparent to folks who are writing tests.Because from the test we should be able to control the total time we are waiting.  See below as an similar example. I will give it more thought and we can discuss how we want to achive that. 

```
/ steps to execute `oc` commands
	s.Step(`^executing "oc (.*)" retrying (\d+) times with wait period of (\d+) seconds$`,
		minishift.executingRetryingTimesWithWaitPeriodOfSeconds)
```

Reply:
@LalatenduMohanty Ack.